File: PerfumeTracker.Server/DTO/PerfumeDTO.cs
using PerfumeTracker.Server.Features.PerfumeRatings;

namespace PerfumeTracker.Server.Dto;

public record PerfumeDto(Guid Id,
	string House,
	string PerfumeName,
	string Family,
	string Parfumeur,
	decimal Ml,
	decimal MlLeft,
	Guid? ImageObjectKey,
	string ImageUrl,
	List<TagDto> Tags,
	bool IsDeleted,
	IReadOnlyList<PerfumeRatingDownloadDto> Ratings,
	int WearCount,
	decimal AverageRating,
	DateTime? LastWorn
	);
public record PerfumeUploadDto(string House, string PerfumeName, string Family, string Parfumeur, decimal Ml, decimal MlLeft, List<TagDto> Tags);

File: perfumetracker.client/src/dto/PerfumeDTO.ts
import { PerfumeRatingDownloadDTO } from "./PerfumeRatingDownloadDTO";
import { TagDTO } from "./TagDTO";

export interface PerfumeDTO {
    id: string;
    house: string;
    perfumeName: string;
    family: string;
    ml: number;
    mlLeft: number;
    imageObjectKey: string;
    imageUrl: string;
    tags: TagDTO[];
    ratings: PerfumeRatingDownloadDTO[];
    wearCount: number | undefined;
}